trie/trie.py

```diff
@@ -455,7 +455,7 @@ def set(self, key, value):
 
         self.root_hash = self._set(self.root_hash, encode_to_bin(key), value)
 
-    def _set(self, node_hash, keypath, value):
+    def _set(self, node_hash, keypath, value, if_delete_subtrie=False):
```

Comment:
I'm confused by the name `if_delete_subtrie`.  It's not clear what it's signaling.  Can you explain or add a comment explaining in human-readable-text what it does.

Reply:
It's used by `delete_subtrie` and call from `delete_subtrie` will set this flag to `True` which indicates if this `_set` action should delete subtrie of a given prefix instead of setting a new value for a given key.

`delete_subtrie` can be written as a standalone function but since it shares most of the logic with `_set`, I reuse the code from `_set` and add a flag to indicate different actions.

I will add a comment for it.